### DISCUSSION TITLE
localePath why need check (typeof window === 'undefined')??

### DISCUSSION THREAD

**Discussion by synasapmob** [i18next/next-i18next#2124] [Q&A] [answered]
I don't understand why need to make as the same below code?, I mean necessary to check 'typeof window' and what happens when we do not check, the difference between `'/locales'` and `require("path").resolve()`

thanks you so read my post, sorry for my bad English !

[https://github.com/i18next/next-i18next/blob/master/examples/simple/next-i18next.config.js](url)
```
  localePath:
    typeof window === 'undefined'
      ? require('path').resolve('./public/locales')
      : '/locales',
```

**Comment by synasapmob** [comment]
@BenjaminIsMyName

**Comment by adrai** [comment, marked as answer]
because on client side the path is different then on server side

**Reply by synasapmob** [reply]
so check 'typeof window' is necessary? or binding?

**Reply by adrai** [reply]
Usually, yes.
But you can try without if the paths are the same.

**Comment by synasapmob** [comment]
resolved the discussion thanks all for that,
if someone has a suggestion for me I still waiting
